akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1338,3 +1338,36 @@ abstract class AbstractOutHandler extends OutHandler
  * (completing when upstream completes, failing when upstream fails, completing when downstream cancels).
  */
 abstract class AbstractInOutHandler extends InHandler with OutHandler
+
+private[akka] trait CallbackWrapper[T] {
```

Comment:
In addition I think it needs scaladoc explaining purpose and how it works if it should be re-usable.